### PR TITLE
Add generation of apiserver-builder executables, .deb and .rpm packages in build/.

### PR DESCRIPTION
**Pull request by metmajer**
This PR creates `apiserver-builder` executables for `darwin-amd64`, `linux-amd64` and `windows-amd64` in `build/` via `make build`:

```
metmajer$ tree build/
build/
├── darwin-amd64
│   └── bin
│       ├── apiregister-gen
│       ├── apiserver-boot
│       └── apiserver-builder-release
├── linux-amd64
│   └── bin
│       ├── apiregister-gen
│       ├── apiserver-boot
│       └── apiserver-builder-release
└── windows-amd64
    └── bin
        ├── apiregister-gen.exe
        ├── apiserver-boot.exe
        └── apiserver-builder-release.exe
```

Additionally, it enables the creation of `.deb` and `.rpm` packages using [jordansissel/fpm](https://github.com/jordansissel/fpm) in `build/` via `make packages`:

```
metmajer$ tree build/
build/
└── linux-amd64
    └── pkg
        ├── apiserver-builder_0.1-alpha.14_amd64.deb
        └── apiserver-builder_0.1-alpha.14_amd64.rpm
```

The version number is now stored in `VERSION` for better version management.

#### Information

```
ubuntu@ubuntu-xenial$ dpkg-deb -I apiserver-builder_0.1-alpha.14_amd64.deb
 new debian package, version 2.0.
 size 5705424 bytes: control archive=579 bytes.
     422 bytes,    11 lines      control
     292 bytes,     4 lines      md5sums
 Package: apiserver-builder
 Version: 0.1-alpha.14
 License: Apache-2.0
 Vendor: kubernetes-incubator
 Architecture: amd64
 Maintainer: The Kubernetes Authors
 Installed-Size: 17381
 Section: default
 Priority: extra
 Homepage: https://github.com/kubernetes-incubator/apiserver-builder
 Description: apiserver-builder implements libraries and tools to quickly and easily build Kubernetes apiservers to support custom resource types.
```

### Installation

Users can install the `.deb` package either using `dpkg -i $name.deb` or `apt-get install -f $name.deb` or `apt install $name.deb`.

```
ubuntu@ubuntu-xenial$ sudo dpkg -i apiserver-builder_0.1-alpha.14_amd64.deb
Selecting previously unselected package apiserver-builder.
(Reading database ... 53918 files and directories currently installed.)
Preparing to unpack apiserver-builder_0.1-alpha.14_amd64.deb ...
Unpacking apiserver-builder (0.1-alpha.14) ...
Setting up apiserver-builder (0.1-alpha.14) ...
```

```
ubuntu@ubuntu-xenial$ ls -la /usr/local/bin/
total 17396
drwxr-xr-x  2 root root    4096 Sep 15 18:31 .
drwxr-xr-x 10 root root    4096 Sep 15 18:31 ..
-rwxr-xr-x  1 root root 7894084 Sep 15 18:30 apiregister-gen
-rwxr-xr-x  1 root root 5598073 Sep 15 18:30 apiserver-boot
-rwxr-xr-x  1 root root 4306566 Sep 15 18:30 apiserver-builder-release
```

```
ubuntu@ubuntu-xenial$ file /usr/local/bin/apiregister-gen
/usr/local/bin/apiregister-gen: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, not stripped
```

#### Information

```
vagrant@fedora$ rpm -qip apiserver-builder_0.1-alpha.14_amd64.rpm
warning: ./apiserver-builder_0.1-alpha.14_amd64.rpm: Header V4 RSA/SHA1 Signature, key ID c1cbfc12: NOKEY
Name        : apiserver-builder
Version     : 0.1_alpha.14
Release     : 1
Architecture: x86_64
Install Date: (not installed)
Group       : default
Size        : 17798723
License     : Apache-2.0
Signature   : RSA/SHA1, Fri 15 Sep 2017 06:38:50 PM UTC, Key ID b8495724c1cbfc12
Source RPM  : apiserver-builder-0.1_alpha.14-1.src.rpm
Build Date  : Fri 15 Sep 2017 06:38:50 PM UTC
Build Host  : martins-macbook-pro.local
Relocations : /
Packager    : The Kubernetes Authors
Vendor      : kubernetes-incubator
URL         : https://github.com/kubernetes-incubator/apiserver-builder
Summary     : apiserver-builder implements libraries and tools to quickly and easily build Kubernetes apiservers to support custom resource types.
Description :
apiserver-builder implements libraries and tools to quickly and easily build Kubernetes apiservers to support custom resource types.
```

```
vagrant@fedora$ rpm -ihv apiserver-builder_0.1-alpha.14_amd64.rpm
warning: apiserver-builder_0.1-alpha.14_amd64.rpm: Header V4 RSA/SHA1 Signature, key ID 8f0e18b6: NOKEY
Preparing...                          ################################# [100%]
Updating / installing...
   1:apiserver-builder-0.1_alpha.14-1 ################################# [100%]
```

```
vagrant@fedora$ ls -la /usr/local/bin/
total 17216
drwxr-xr-x.  3 root root    4096 Sep 15 20:20 .
drwxr-xr-x. 12 root root    4096 Nov 15  2016 ..
-rwxr-xr-x.  1 root root 7888579 Sep 15 20:18 apiregister-gen
-rwxr-xr-x.  1 root root 5530494 Sep 15 20:18 apiserver-boot
-rwxr-xr-x.  1 root root 4197066 Sep 15 20:18 apiserver-builder-release
```

With that, `.deb` and `.rpm` packages can now be attached to a release on GitHub. If you like, I can also include the generation of source and binary `.tar.gz` distributables.

### CODE REVIEW

**Comment by metmajer** [comment]
Oh my... I've meant to squash 2 consecutive commits and now ended up in a mess of 8 commits. The change set is fully intact though. Sorry.

**Comment by pwittrock** [comment]
Try just doing a rebase onto head

File: Makefile [hunk 1]
```diff
@@ -20,3 +37,70 @@ test:
...
+	rm -rf $(BUILD_DIR)
+
+.PHONY: build
+build: clean build-darwin-amd64 build-linux-amd64 build-windows-amd64
+
+.PHONY: build-darwin-amd64
+build-darwin-amd64: build-apiregister-gen-darwin-amd64 build-apiserver-boot-darwin-amd64 build-apiserver-builder-release-darwin-amd64
+
+.PHONY: build-apiregister-gen-darwin-amd64 build-apiserver-boot-darwin-amd64 build-apiserver-builder-release-darwin-amd64
+build-apiregister-gen-darwin-amd64 build-apiserver-boot-darwin-amd64 build-apiserver-builder-release-darwin-amd64: build-%-darwin-amd64:
```
Comment (pwittrock) [inline Makefile]: I think we will also want to build the vendored bins for the code generator, as well as include the vendored libraries.

apiserver-builder-release does these steps and produces a tar.gz, so maybe this just needs to run the release script and repackage the tar?

https://github.com/kubernetes-incubator/apiserver-builder/blob/master/cmd/apiserver-builder-release/main.go

**Comment by metmajer** [comment]
I was doing `git rebase -i HEAD~$n` where $n was the number of commits that showed up in the pull request. Squashing these commits into my first commit always resulted in generating only more commits. This branch became totally screwed, so I thought it was best to close and recreate. Sorry for the mess.

**Pull request closed without merging**